### DISCUSSION TITLE
Minimal custom build for ML-KEM

### DISCUSSION THREAD

**Discussion by jiep** [randombit/botan#4431] [General]
First of all, thanks for this awesome library!

I want to compile a custom Botan library, including only one ML-KEM set parameter, e.g. ML-KEM-768. I tried to include modules with options  `--minimized-build --enable-modules`. Including option `ml_kem` adds all 3 parameters for ML-KEM, but I don't find any option to include just one parameter set. Is this possible to do that? Is any plans to add it soon?

Thanks in advance. 

Regards.

**Comment by randombit** [comment]
I don't think we would want to split things up with this level of granularity, mostly because here the savings in code size seem quite minimal especially when compared to the additional complexity that would be brought in. For every case where some functionality is a distinct module, we have to consider everywhere else in the library if it does or does not exist.

**Comment by jiep** [comment]
Thanks for the quick response!
